src/DynamoWebServer/Messages/MessageHandler.cs

```diff
@@ -465,6 +473,44 @@ private string GetValue(NodeModel node)
             return data;
         }
 
+        private object GenerateWatchData(MirrorData data)
+        {
+            if (data.IsCollection)
+            {
+                var list = new List<object>();
+                var elements = data.GetElements();
+
+                foreach (var mirrorData in elements)
+                {
+                    list.Add( mirrorData.IsCollection
+                            ? GenerateWatchData(mirrorData)
+                            : GetSingleObjectPreview(mirrorData));
+                }
+
+                return list.ToArray();
+            }
+
+            return GetSingleObjectPreview(data);
+        }
+
+        private object GetSingleObjectPreview(MirrorData data)
+        {
+            if (data.Data != null)
+            {
+                double number;
+                return double.TryParse(data.StringData, out number)
+                    ? data.Data
+                    : data.Data.ToString();
```

Comment:
You mean when you did a `double.TryParse` that `data.StringData` has actual contents like `\"42\"`? If that is the case `double.TryParse` will fail, right?